tests/test_cache_control.py

```diff
@@ -115,6 +115,22 @@ def test_cache_response_no_store(self):
         cc.cache_response(self.req(), resp)
         assert not cc.cache.get(cache_url)
 
+    def test_cache_response_no_store_with_etag(self):
```

Comment:
I think this test can be reduced to this:

    def test_cache_response_no_store_with_etag(self, cc):
        resp = self.resp({'cache-control': 'no-store', 'ETag': 'jfd9094r808'})
        cc.cache_response(self.req(), resp)

        assert not cc.cache.set.called